check/src/kindcheck.rs

```diff
@@ -139,14 +139,9 @@ impl<'a> KindCheck<'a> {
                 Ok((gen.kind.clone(), Type::generic(gen)))
             }
             Type::Variable(_) => panic!("kindcheck called on variable"),
+            Type::Builtin(BuiltinType::Array) |
```

Comment:
Array and function do not have the same kind (`Type -> Type` vs `Type -> Type -> Type`). If you can add a test to catch this as well that would be great!